tools/release.sh

```diff
@@ -125,6 +126,7 @@ pip install -U pip
 pip install \
   --no-cache-dir \
   --extra-index-url http://localhost:$PORT \
+  --constraint constraints.txt \
```

Comment:
When I hit this line locally the script crashes with:
```
ERROR: Invalid requirement: 'Warning: The lock file is not up to date with the latest changes in pyproject.toml. You may be getting outdated dependencies. Run update to update them.' (from line 1 of constraints.txt)
```
Also, can we move `constraints.txt` out of the `dist.$version` directory? That currently only contains things we upload to PyPI and I think it'd be nice to keep it that way.

Reply:
Indeed the lock file was outdated. This is fixed. I also moved the generated constraints.txt file to upper level. I also added it to `.gitignore` so that it is never committed, but I can also modify the `release.sh` script to put the file in a disposable temporary directory if you prefer.